chain/client/src/client.rs

```diff
@@ -174,6 +174,8 @@ pub struct Client {
     upgrade_schedule: ProtocolUpgradeVotingSchedule,
     /// Produced optimistic block.
     last_optimistic_block_produced: Option<OptimisticBlock>,
+    /// Cached precomputed set of TIER1 accounts to which Optimistic Block should be sent.
+    optimistic_block_targets_cache: Option<(EpochId, Arc<Vec<AccountId>>)>,
```

Comment:
Is it the targets just within this epoch or this and next? Either way can you mention that in the comment?

Reply:
I'd be tempted to try to remove the Arc and make it use references - though I'm not sure if that would be any better.